crates/ruff/src/rules/isort/categorize.rs

```diff
@@ -239,34 +274,46 @@ impl KnownModules {
                 .rev()
             {
                 let submodule = &module_name[0..i];
-                if self.first_party.contains(submodule) {
-                    return Some((ImportType::FirstParty, Reason::KnownFirstParty));
-                }
-                if self.third_party.contains(submodule) {
-                    return Some((ImportType::ThirdParty, Reason::KnownThirdParty));
-                }
-                if self.local_folder.contains(submodule) {
-                    return Some((ImportType::LocalFolder, Reason::KnownLocalFolder));
-                }
-                if self.standard_library.contains(submodule) {
-                    return Some((ImportType::StandardLibrary, Reason::ExtraStandardLibrary));
+                if let Some(result) = self.categorize_submodule(submodule) {
+                    return Some(result);
                 }
             }
             None
         } else {
             // Happy path: no submodules, so we can check the module base and be done.
             let module_base = module_name.split('.').next().unwrap();
-            if self.first_party.contains(module_base) {
-                Some((ImportType::FirstParty, Reason::KnownFirstParty))
-            } else if self.third_party.contains(module_base) {
-                Some((ImportType::ThirdParty, Reason::KnownThirdParty))
-            } else if self.local_folder.contains(module_base) {
-                Some((ImportType::LocalFolder, Reason::KnownLocalFolder))
-            } else if self.standard_library.contains(module_base) {
-                Some((ImportType::StandardLibrary, Reason::ExtraStandardLibrary))
-            } else {
-                None
-            }
+            self.categorize_submodule(module_base)
+        }
+    }
+
+    fn categorize_submodule(&self, submodule: &str) -> Option<(ImportSection, Reason)> {
```

Comment:
The custom name is used during formatting, it determines the order of the sections within the block. It could probably be made to work with a smaller type for the section by mapping section names to IDs when the configuration is read, but I didn't think it would be worth the additional complexity.